org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java

```diff
@@ -964,6 +987,185 @@ public void recyclableThreads(ThreadReference thread) {
+                        newValue = setArrayValue(array, eleType, Integer.parseInt(arguments.name), arguments.value, options);
+                    } else {
+                        if (StringUtils.isBlank(belongToClass)) {
+                            Field field = currentObj.referenceType().fieldByName(name);
+                            if (field != null) {
+                                if (field.isStatic()) {
+                                    newValue = this.setStaticFieldValue(currentObj.referenceType(), field,
+                                            arguments.name, arguments.value, options);
+                                } else {
+                                    newValue = this.setObjectFieldValue(currentObj, field, arguments.name,
+                                            arguments.value, options);
+                                }
+                            } else {
+                                throw new IllegalArgumentException(
+                                        String.format("SetVariableRequest: Variable %s cannot be found.", arguments.name));
+                            }
+                        } else {
+                            newValue = setFieldValueWithConflict(currentObj, currentObj.referenceType().allFields(),
+                                    name, belongToClass, arguments.value, options);
+                        }
+                    }
+                } else {
+                    throw new IllegalArgumentException(
+                            String.format("SetVariableRequest: Variable %s cannot be found.", arguments.name));
+                }
+            } catch (IllegalArgumentException | AbsentInformationException | InvalidTypeException
+                    | UnsupportedOperationException | ClassNotLoadedException e) {
+                return new Responses.ErrorResponseBody(convertDebuggerMessageToClient(e.getMessage()));
+            }
+            int referenceId = getReferenceId(thread, newValue, showStaticVariables);
+
+            int indexedVariables = 0;
+            if (newValue instanceof ArrayReference) {
+                indexedVariables = ((ArrayReference) newValue).length();
+            }
+            return new Responses.SetVariablesResponseBody(
+                    this.variableFormatter.typeToString(newValue == null ? null : newValue.type(), options), // type
+                    this.variableFormatter.valueToString(newValue, options), // value,
+                    referenceId, indexedVariables);
+
+        }
+
+        private Value setValueProxy(Type type, String value, SetValueFunction setValueFunc, Map<String, Object> options)
+                throws ClassNotLoadedException, InvalidTypeException {
+            IValueFormatter formatter = getFormatterForModification(type, options);
+            Value newValue = formatter.valueOf(value, type, options);
+            setValueFunc.apply(newValue);
```

Comment:
Can it be another member function of this class? yes, it can be.

There will be a various of setValue functions, stackFrame.setValue, array.setValue,  object.setValue,  classType.setValue, all of them are using different parameters and throwing same exceptions, using a SetValueFunction as the signature to keep these code reused but kepts there exceptions, otherwise you need to catch the InvalidTypeException, ClassNotLoadedException exceptions, but have no clean way of reporting to the callers.  

Object:void setValue(Field field, Value value)
        throws InvalidTypeException, ClassNotLoadedException;
Array:   void setValue(int index, Value value)
            throws InvalidTypeException,
                   ClassNotLoadedException;
StackFrame:void setValue(LocalVariable variable, Value value)
        throws InvalidTypeException, ClassNotLoadedException;
ClassType: void setValue(Field field, Value value) throws InvalidTypeException, ClassNotLoadedException